### PR TITLE
factored out make_rest_server_debug/prod

### PR DESCRIPTION
**Pull request by zyxue**
The idea is to factor parts out of main to make it more modular and readable.

If this PR works out. will factor out `_make_rest_metrics_server` and `_make_grpc_server` next.

Running `make test` locally seems flaky, even on master.

### CODE REVIEW

**Comment by seldondev** [comment]
Hi @zyxue. Thanks for your PR.

I'm waiting for a [SeldonIO](https://github.com/orgs/SeldonIO/people) or [todo](https://github.com/orgs/todo/people) member to verify that this patch is reasonable to test. If it is, they should reply with `/ok-to-test` on its own line. Until that is done, I will not automatically test new commits in this PR, but the usual testing commands by org members will still work. Regular contributors should [join the org](https://github.com/orgs/todo/people) to skip this step.

Once the patch is verified, the new status will be reflected by the `ok-to-test` label.

I understand the commands that are listed [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=SeldonIO%2Fseldon-core).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository.
</details>

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **axsaucedo**
You can assign the PR to them by writing `/assign @axsaucedo` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=SeldonIO%2Fseldon-core).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: python/seldon_core/microservice.py [hunk 1]
```diff
@@ -353,6 +353,102 @@ def parse_args():
     return parser.parse_known_args()
 
 
+def _make_rest_server_debug(
+    user_object: Any,
+    seldon_metrics: SeldonMetrics,
+    args: argparse.Namespace,
+    *,
```
Comment (axsaucedo) [inline python/seldon_core/microservice.py:360]: Here there is an expected number of args and kwargs so no need to allow anonymous args

File: python/seldon_core/microservice.py [hunk 2]
```diff
@@ -353,6 +353,102 @@ def parse_args():
...
+        )
+
+    return server
+
+
+def _make_rest_server_prod(
+    user_object: Any,
+    seldon_metrics: SeldonMetrics,
+    args: argparse.Namespace,
+    *,
```
Comment (axsaucedo) [inline python/seldon_core/microservice.py:400]: Same here

**Review by axsaucedo — COMMENTED** [review]
Thank you for the contribution @zyxue - changes look reasonable, I just added a small change to remove anonymous args, but overall should make it more readable, thanks.

File: python/seldon_core/microservice.py (hunk 1 above)
Comment (zyxue) [reply python/seldon_core/microservice.py:360]: this is not about annoymous args, but keyword-only args, https://peps.python.org/pep-3102/

Comment (zyxue) [reply python/seldon_core/microservice.py:360]: but I can remove `*`, if it's considered unnecessary.

Comment (axsaucedo) [reply python/seldon_core/microservice.py:360]: Yeah, anonimous args as allowing for any further args, but not sure why this is needed so best remove

**Comment by axsaucedo** [comment]
Integration tests pass, can merge once the updates above are carried out

image

**Comment by zyxue** [comment]
@axsaucedo , mind take another look, please?

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *[axsaucedo](https://github.com/SeldonIO/seldon-core/pull/4268#issuecomment-1256260967 "Approved")*

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=SeldonIO%2Fseldon-core).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)~~ [axsaucedo]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**